### ISSUE TITLE
[Fleet] Updating cluster group fails with error

### ISSUE THREAD

**Issue by janeczku** [rancher/dashboard#1926]
1. Create Cluster Group specifying at least one cluster selector rule
2. Cluster Group -> Edit as form
3. Click Save

**Result**

Save fails with following error:

> ClusterGroup.fleet.cattle.io "test-group-1" is invalid: spec.selector.matchExpressions.values: Invalid value: "string": spec.selector.matchExpressions.values in body must be of type array: "string"

<img width="1417" alt="Screenshot 2020-11-27 at 15 08 12" src="https://user-images.githubusercontent.com/3813921/100457495-6b7bdb80-30c2-11eb-8102-b66fe205d4cf.png">

Rancher 2.5.2

**Comment by vincent99** [comment]
@janeczku Can you reproduce this in master (or 2.5.4-rcX)?  It seems ok to me now.

**Comment by deniseschannon** [comment]
@izaac Can you test with the latest?

**Comment by jiaqiluo** [comment]
The bug is reproduced in v2.5-head fc09f27, and there are 2 related bugs. 

We can reproduce it by following those steps:

- Add a `cluster group` 
![image](https://user-images.githubusercontent.com/6218999/103699699-8f073100-4f60-11eb-915a-d8fa6f8381d0.png)

- If we go to edit the `cluster group` as config now, there will be no selector showing in the UI, but we can save it and nothing will be changed.  <--- **this is another bug we need to fix** 
![image](https://user-images.githubusercontent.com/6218999/103699788-b3fba400-4f60-11eb-90a6-9bdc831f46c5.png)

<img width="1383" alt="Screen Shot 2021-01-05 at 2 18 11 PM" src="https://user-images.githubusercontent.com/6218999/103699926-e907f680-4f60-11eb-8b99-fa1373fd78fa.png">

Now let's try to trigger the bug this issue mentioned:
- Click on the name of the cluster group to open its detailed page
- Choose to edit it as config
<img width="1370" alt="Screen Shot 2021-01-05 at 2 21 47 PM" src="https://user-images.githubusercontent.com/6218999/103700287-7fd4b300-4f61-11eb-82f9-5cee8f0756a1.png">

- Now we should see the selector in the UI. Click on save will trigger the bug 
![image](https://user-images.githubusercontent.com/6218999/103700394-a09d0880-4f61-11eb-821e-5535bb62e04f.png)

**Comment by vincent99** [comment]
The important bit here is that the ones that are missing are "simple" key=value selectors.  Anything that can be expressed as a `matchLabels` entry gets converted to one, so only the more complicated entries get stored into `matchExpressions`.

Then on edit only the `matchExpressions` were getting restored.  Last time I tried it was with something that wasn't a "simple" entry, so it restored fine.  In your example that matchExpressions is empty and matchLabels is ignored, so nothing is loaded to edit (and then an empty row is added independently, like it is on create).

Changed to load both matchLabels & matchExpressions.
